Example-Swift/PhotoViewerCoordinator.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  PhotoViewerCoordinator.swift
+//  NYTPhotoViewer
+//
+//  Created by Chris Dzombak on 2/2/17.
+//  Copyright © 2017 NYTimes. All rights reserved.
+//
+
+import NYTPhotoViewer
+
+/// Coordinates interaction between the application's data layer and the photo viewer component.
+final class PhotoViewerCoordinator: NYTPhotoViewerDataSource {
+    let slideshow: [NYTPhotoBox]
+    let provider: PhotosProvider
+
+    lazy var photoViewer: NYTPhotosViewController = {
+        return NYTPhotosViewController(dataSource: self)
+    }()
+
+    init(provider: PhotosProvider) {
+        self.provider = provider
+        self.slideshow = provider.fetchDemoSlideshow().map { NYTPhotoBox($0) }
+        self.fetchPhotos()
+    }
+
+    func fetchPhotos() {
+        for box in slideshow {
+            provider.fetchPhoto(named: box.value.name, then: { [weak self] (result) in
+                box.image = result
+                self?.photoViewer.update(box)
+            })
+        }
+    }
+
+    // MARK: NYTPhotoViewerDataSource
+
+    @objc
+    var numberOfPhotos: NSNumber? {
```

Comment:
Small thing, but since it's sample code that may be copied/pasted: Swift's `Int` is boxed to `NSNumber` when needed, so you could
```swift
    @objc
    var numberOfPhotos: Int? {
        return slideshow.count
    }
```